sequence_processing_pipeline/QCJob.py

```diff
@@ -92,57 +114,37 @@ def _filter(self, filtered_directory, empty_files_directory,
-            self._clear_trim_files()
-            lines_per_split = int((
-                len(fastq_files) + split_count - 1) / split_count)
-            trim_files = self._generate_trim_files(
-                fastq_files, lines_per_split)
-            if len(trim_files) != split_count:
-                logging.warning("The number of trim files does not equal the "
-                                "number of expected files.")
-
-            script_path = self._generate_job_script(project['Sample_Project'],
-                                                    project['ForwardAdapter'],
-                                                    project['ReverseAdapter'],
-                                                    project['HumanFiltering'])
-
-            pbs_job_id = self.qsub(script_path, None, None)
+            project_name = project['Sample_Project']
+            pbs_job_id = self.qsub(self.script_paths[project_name], None, None)
             logging.debug(f'QCJob {pbs_job_id} completed')
-
-            source_dir = join(self.products_dir, project['Sample_Project'])
+            source_dir = join(self.products_dir, project_name)
             filtered_directory = join(source_dir, 'filtered_sequences')
             empty_files_directory = join(source_dir, 'zero_files')
             self._filter(filtered_directory, empty_files_directory,
                          self.minimum_bytes)
 
-    def _generate_trim_files(self, fastq_files, split_count):
-        def _chunk_list(some_list, n):
-            # taken from https://bit.ly/38S9O8Z
-            for i in range(0, len(some_list), n):
-                yield some_list[i:i + n]
-
-        new_files = []
-        count = 0
-        for chunk in _chunk_list(fastq_files, split_count):
-            trim_file_name = '%s%d' % (self.trim_file, count)
-            trim_file_path = join(self.run_dir, trim_file_name)
+    def _generate_trim_files(self, project_name, fastq_files, split_count):
+        chunk_list = [fastq_files[i:i + split_count] for i in
+                      range(0, len(fastq_files), split_count)]
+        trim_files = []
+
+        for count, chunk in enumerate(chunk_list):
+            trim_file_path = join(self.run_dir,
+                                  f'split_file_{project_name}_{count}')
             with open(trim_file_path, 'w') as f:
                 for line in chunk:
                     f.write("%s\n" % line)
-            new_files.append(trim_file_path)
-            count += 1
+            trim_files.append(trim_file_path)
 
-        return new_files
+        return trim_files
 
     def _clear_trim_files(self):
         # remove all files with a name beginning in self.trim_file.
         # assume cleaning the entire run_dir is overkill, but won't
         # hurt anything.
         for root, dirs, files in walk(self.run_dir):
             for some_file in files:
-                if self.trim_file in some_file:
+                if 'split_file_' in some_file:
```

Comment:
If this name changes, it would also be needed to change here, correct? If yes, then it would be better to make this a variable or part of the configuration.